lm_eval/tasks/ifeval/instructions_util.py

```diff
@@ -15,33 +15,40 @@
 """Utility library of instructions."""
 
 import functools
+import os
 import random
 import re
+import time
+from importlib.metadata import version
 
 import immutabledict
 import nltk
-import pkg_resources
-from packaging import version
+from packaging.version import parse as parse_version
 
 
 # Downloading 'punkt' with nltk<3.9 has a remote code vuln.
 # see  https://github.com/EleutherAI/lm-evaluation-harness/issues/2210
 # and https://github.com/nltk/nltk/issues/3266
 # for more information.
 NLTK_MIN_VERSION = "3.9.1"
+RANK = os.environ.get("LOCAL_RANK", "0")
 
 
 def download_nltk_resources():
     """Download 'punkt' if not already installed"""
-    nltk_version = pkg_resources.get_distribution("nltk").version
     assert (
-        version.parse(nltk_version) >= version.parse(NLTK_MIN_VERSION)
+        (nltk_version := parse_version(version("nltk")))
+        >= parse_version(NLTK_MIN_VERSION)
     ), f"`nltk` version {nltk_version} is not >= {NLTK_MIN_VERSION}. Please update `nltk` before proceeding--older versions are vulnerable to a remote code execution vulnerability."
 
     try:
         nltk.data.find("tokenizers/punkt_tab")
     except LookupError:
-        nltk.download("punkt_tab")
+        if RANK == "0":
+            nltk.download("punkt_tab")
+            print("Downloaded punkt_tab")
+        else:
+            time.sleep(5)
```

Comment:
This isn't really necessary as the code runs before generations. But couldn't hurt.

Reply:
suggestion: if sleep is not needed we should not add it.